src/rules/noShadowedVariableRule.ts

```diff
@@ -48,6 +50,12 @@ class NoShadowedVariableWalker extends Lint.BlockScopeAwareRuleWalker<ScopeInfo,
         this.visitBlock(node.block);
     }
 
+    public visitFunctionType(node: ts.FunctionOrConstructorTypeNode) {
+        this.isInFunctionType = true;
+        super.visitFunctionType(node);
+        this.isInFunctionType = false;
+    }
```

Comment:
instead of the `isInFunctionType` flag, what if we just don't call `super.visitFunctionType(node)` here? that should prevent the walker from checking nested parameter declarations.